DeviceSimulation/SimulationTests.cs

```diff
@@ -29,5 +40,190 @@ public void Should_Return_OK_Status()
+        {
+
+            var request1 = new HttpRequest(DS_ADDRESS + "/simulations/1");
+            var response1 = this.httpClient.GetAsync(request1).Result;
+            Assert.Equal(HttpStatusCode.OK, response1.StatusCode);
+            JObject JsonResponse = JObject.Parse(response1.Content);
+
+            string ETag = (string)JsonResponse["ETag"];
+
+            var simulation2 = "{" + $"'ETag': '{ETag}' ,'Enabled': true" + "}";
+            var byteArray = Encoding.ASCII.GetBytes(simulation2);
+
+            var request2 = (HttpWebRequest)WebRequest.Create("http://localhost:9003/v1/simulations/1");
+            request2.Method = "PATCH";
+            request2.ContentLength = byteArray.Length;
+            request2.ContentType = "application/json";
+            Stream dataStream = request2.GetRequestStream();
+            dataStream.Write(byteArray, 0, byteArray.Length);
+            dataStream.Close();
+            var response2 = (HttpWebResponse)request2.GetResponse();
+            Assert.Equal(HttpStatusCode.OK, response2.StatusCode);
+
+            var request = new HttpRequest(DS_ADDRESS + "/simulations/1");
+            var response = this.httpClient.GetAsync(request).Result;
+            JObject jsonResponse = JObject.Parse(response.Content);
+
+            Assert.True((bool)jsonResponse["Enabled"]);
+        }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public void Should_Stop_Given_Simulation()
+        {
+            var request1 = new HttpRequest(DS_ADDRESS + "/simulations/1");
+            var response1 = this.httpClient.GetAsync(request1).Result;
+            Assert.Equal(HttpStatusCode.OK, response1.StatusCode);
+            JObject JsonResponse = JObject.Parse(response1.Content);
+
+            string ETag = (string)JsonResponse["ETag"];
+
+            var simulation2 = "{" + $"'ETag': '{ETag}' ,'Enabled': false" + "}";
+            var byteArray = Encoding.ASCII.GetBytes(simulation2);
+
+            var request2 = (HttpWebRequest)WebRequest.Create(PATCH_SIMULATION);
+            request2.Method = "PATCH";
+            request2.ContentLength = byteArray.Length;
+            request2.ContentType = "application/json";
+            Stream dataStream = request2.GetRequestStream();
+            dataStream.Write(byteArray, 0, byteArray.Length);
+            dataStream.Close();
+            var response2 = (HttpWebResponse)request2.GetResponse();
+            Assert.Equal(HttpStatusCode.OK, response2.StatusCode);
+
+            var request = new HttpRequest(DS_ADDRESS + "/simulations/1");
+            var response = this.httpClient.GetAsync(request).Result;
+            JObject jsonResponse = JObject.Parse(response.Content);
+
+            Assert.False((bool)jsonResponse["Enabled"]);
+        }
+
+        private void Should_Delete_Existing_Simulation()
+        {
+            var request = new HttpRequest(DS_ADDRESS + "/simulations/1");
+            var response = this.httpClient.DeleteAsync(request).Result;
```

Comment:
What happens if there is no running simulation?

Reply:
In all test cases, I am deleting the current simulation (if any) and creating a new one for that particular unit test. So, we will not run into this case.